### ISSUE TITLE
Verification Info: As an Industry Reporter, I want to fill out the SFO verification form

### ISSUE THREAD

**Issue by suhafa** [bcgov/cas-reporting#199]
#### Description:

Before signing off and submitting an Annual Report, the verification page will need to be filled out. A verification statement will also need to uploaded as part of this step. (uploaded in the attachments page)
All Regulated Operations must fill out this page. 
Reporting Operations who have emissions above 25,000 T CO2e will also need to do this step (reporting operations under 25,000T will not). 

[SFO Wireframe](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=7826-19093&t=IlPsMexuLNx5JIwR-4)

[Form fields](
https://bcgov.sharepoint.com/:x:/r/teams/00608-CPRInternal/_layouts/15/doc2.aspx?sourcedoc=%7BB6613650-00EB-422F-A103-41608CD0E2E6%7D&file=BCIERS%20Reporting%20Fields%20v2.xlsx&action=default&mobileredirect=true)

**Note:** [LFO Wireframe](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=17438-4851&t=kkLCs25GtG7LwPro-4) will be handled using ticket [410](https://github.com/bcgov/cas-reporting/issues/410)

#### Acceptance Criteria:
Given that I am a REGULATED Operation 
When I have filled out all the required forms for emissions and production, and have seen my final Compliance Summary
Then I need to fill out the Verification form page 

~Given I am a REGULATED Operations
When I am on the attachments page
Then I MUST include a verification statement upload.~ This will be bcgov/cas-registration#371 

Given I am a REPORTING Operation who has attributable emissions greater than or equal to 25,000 T CO2e (not including CO2 from biomass listed in item 1 of Schedule C)
When I have filled out all the required emissions forms and have seen my Emissions Summary page
Then I need to fill out the verification form page. 
*Note: Reporting operations do NOT fill in the production/compliance data or have a compliance summary

Given I am a REPORTING Operation who has attributable emissions greater than or equal to 25,000 T CO2e (not including CO2 from biomass listed in item 1 of Schedule C)
When I am on the attachments page
Then I MUST include a verification statement upload.

Dropdown lists
Accredited by = "ANAB" or "SCC"
Scope of verification = "B.C. OBPS Annual Report" or "Supplementary Report" or "Corrected Report"

Site visited = [show list of registered facilities for the operation] and allow selection of one or more facilities, or "None", or "Other".
- IF "Other", then show a textbox field "Please indicate the site visited, its location, and type of site visit conducted on this site."
- IF "None", then don't show the type of site visit field

#### Pre-req Checklist:

- The conditions of being a regulated operation or a reporting operation with emissions over 25,000 T CO2e are checked when all emissions are filled out
- If either condition is true, then the verification form page is rendered matching the wireframe
#### Post-req:
-  The ~verification~ attachment form allows the user to upload a verification statement and saves to the db when submitted (see ticket bcgov/cas-registration#200)

#### Development Checklist:

**BE**
- [x] create model `bc_obps/reporting/models/report_verification.py`
- [x] add model reference to `bc_obps/reporting/models/__init__.py`
- [x] create model migration: `bc_obps$ make migrations`
- [x] create model based schema `bc_obps/reporting/schema/report_verification.py`
- [x] create service to save to database `bc_obps/reporting/service/report_verification_service.py`
- [x] create endpoint to POST payload `bc_obps/reporting/api/report_verification.py`
- [x] create endpoint to GET facilities by version id `bc_obps/reporting/api/report_facilities.py`
- [x] create service to return facilities by version id `bc_obps/reporting/service/report_facilities_service.py`
- [x] add endpoint references to `bc_obps/reporting/api/__init__.py`
- [x] update common model test `bc_obps/registration/tests/models/test_user.py`
- [x] update common endpoint test `bc_obps/common/tests/endpoints/auth/test_endpoint_permissions.py`
- [x] update test for model `bc_obps/reporting/tests/models/test_report_version`
- [x] add test for model `bc_obps/reporting/tests/models/test_report_verification.py`
- [x] add test for service `bc_obps/reporting/tests/service/test_report_facilities_service.py`
- [x] add test for service `bc_obps/reporting/tests/service/test_report_verification_service.py`
- [x] add test for api `bc_obps/reporting/tests/api/test_report_facilities_api.py`
- [x] add test for api `bc_obps/reporting/tests/api/test_report_verification_api.py`

**FE**
- [x] add app route `bciers/apps/reporting/src/app/bceidbusiness/industry_user/reports/[version_id]/verification/page.tsx`
- [x] add app route `bciers/apps/reporting/src/app/bceidbusiness/industry_user_admin/reports/[version_id]/verification/page.tsx`
- [x] add app route common page `bciers/apps/reporting/src/app/components/signOff/verification/VerificationPage.tsx`
- [x] add fn for dynamic schema `bciers/apps/reporting/src/app/components/signOff/verification/createVerificationSchema.ts`
- [x] add verification form `bciers/apps/reporting/src/app/components/signOff/verification/VerificationForm.tsx`
- [x] add form schemas `bciers/apps/reporting/src/data/jsonSchema/signOff/verification/verification.ts`
- [x] add vitests
- [x] add vitest helpers

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

- Business Area creating a template for this so we're blocked on this page. 
bcgov/cas-registration#200 blocked by this

**Comment by NicoleGovvy** [comment]
From verification meeting notes:

- Once the template is finished (or closer to being finished) there will be a more solid idea of what to include here.
- Some items in the UI may not be needed or can be re-worked
- Desire to have as much automated/selected (like selecting a site for a visit) to reduce the risk of errors
- What is valuable to the business area for analytics?
- What can be a data collection point that's helpful for CAS?

**Comment by patriciarussellCAS** [comment]
Here's another one for devs - any chance someone could tackle the Dev Checklist for this one prior to next week's Refinement? 
@BallardRobinett @ayeshmcg @joshgamache ?

**Comment by BallardRobinett** [comment]
I can do the dev checklist for this one

**Comment by rdromey** [comment]
Will revisit in next week's refinement to see if we're actually going to do it after discussion with BA

**Comment by patriciarussellCAS** [comment]
@shon-button just tagged you in figma. There has been some updated notes on these fields in the BCIERS Reporting Fields v2 spreadsheet. We may need to clarify how to manage the "type of site visit" selection, given they can select more than one facility in "site visited". Thoughts on how to do this are appreciated. 
cc: @dleard @pbastia

**Comment by pbastia** [comment]
We do have a multi-select component (for example on the operation data review page) that we could re-use here, asking reporters to select from their facilities' list

**Comment by patriciarussellCAS** [comment]
@shon-button I've updated this ticket to include a link to the LFO wireframe, as there are additional considerations we need to consider for them. Sorry for the last minute updates to this form! Happy to discuss.

**Comment by shon-button** [comment]
LFO functionality split from SFO functionality in [ticket 410](https://github.com/bcgov/cas-reporting/issues/410)

**Comment by shon-button** [comment]
- Login `cas-dev`
https://cas-bciers-frontend-dev.apps.silver.devops.gov.bc.ca
- Navigate to Reporting\Reports:
https://cas-bciers-frontend-dev.apps.silver.devops.gov.bc.ca/reporting/reports
- Click row `Operation 3` button `Continue`
- Change url from `/review-operator-data` to `/verification?reports_title=Operation%203`
**Expected results:** 
   - SFO report verification form is displayed
   - Section header displays active for `Sign-off & Submit`
   - Multistep displays active for `Verification`
   - Form has button `Cancel`
   - Form has button `Save and Continue`
![image](https://github.com/user-attachments/assets/758ee252-cd0e-4a54-9153-13eb554e9945)
- Ensure form works as expected
[Example Testing Steps](https://github.com/bcgov/cas-registration/pull/2496)

**Comment by patriciarussellCAS** [comment]
Thanks Shon - this is looking great!
I noticed one minor issue, which I think is due to the wireframe not being totally clear, so apologies for that. 
The fields: 
Were there any threats to independence noted Y/N
Verification conclusion (dropdown menu)
Both need to be displayed when a facility is selected, not just when they select OTHER from "site visited" 

The other additional fields for other:
Please indicate the site visited & Geographic coordinates are correct. 

@zoeyli-46 @shon-button

**Comment by shon-button** [comment]
> Were there any threats to independence noted Y/N
> Verification conclusion (dropdown menu)

@patriciarussellCAS 

Can you confirm which (if any) of the fields below are to display when:
Sites visited* = **None**

- [ ] Type of site visit*
- [ ] Were there any threats to independence noted*
- [ ] Verification conclusion*

**Comment by patriciarussellCAS** [comment]
Looks like we got confirmation from the BA, but to summarize: 
For **every** "Site visited" option (Facility X, Other, None) we need these datafields: 
Were there any threats to independence noted*
Verification conclusion*

If it's Facility [x] or Other we also need to include: 
Type of site visit: Virtual or In Person

If Other was selected, then we also need to show 2 additional fields: 
"please indicate the site visited"
geographic location of site

**Comment by patriciarussellCAS** [comment]
@vivanovbc are you still working on this, or should it go back to @shon-button ?

**Comment by vivanovbc** [comment]
@patriciarussellCAS latest issue is fixed and merged, I believe it can go back to @shon-button or closed

**Comment by fviduya** [comment]
Hi @vivanovbc, I have tested this in the DEV environment using the bc-cas-dev role. Here are the results and some things I noticed:

Steps to Test:
1. Log in as ```bc-cas-dev```.
2. Navigate to Reporting/Reports by clicking Submit Annual Report(s) under Reporting Tile.
3. Click Start/Continue on Operation 3.
4. Change url from ```/review-operator-data``` to ```/verification?reports_title=Operation%203```

<img width="468" alt="Image" src="https://github.com/user-attachments/assets/7743d2a8-69a4-42c5-ae15-4ebcbcbfa957" />

Expected Result:

[x] SFO report verification form is displayed
[x] Section header displays active for ```Sign-off & Submit```
[x] Multistep displays active for ```Verification```
[ ] Form has button ```Cancel``` **(Actual Result: Form has buttons ```Back``` and ```Save``` which is what I see in Figma too)**
[x] Form has button ```Save and Continue```
[x] ```Were there any threats to independence noted*``` and ```Verification conclusion*``` is displayed
[x] ```Type of Visit: Virtual or In-person``` is displayed when Facility X or Other is selected in ```Site visited```
<img width="468" alt="Image" src="https://github.com/user-attachments/assets/b78dc7ba-6550-45b4-9ae7-13a9507fa269" />
<img width="468" alt="Image" src="https://github.com/user-attachments/assets/03ed12c8-76c2-4c5a-a43a-19c149611a33" />

[x] ```Please indicate the site visited``` and ```Geographic coordinates of site``` is displayed when Other is selected in ```Site visited```
<img width="468" alt="Image" src="https://github.com/user-attachments/assets/cbd13bb0-2b78-4c0d-a248-ae87f4b4f36f" />


A few things I noticed:
- Figma wireframe is not updated to show these changes (?)
- Shouldn't it be **Site** visited and not **Sites**? The current one is in plural form.
- After clicking ```Save and Continue```, page navigates to ```Attachments``` which is expected. When I click on ```Verification``` again from there, these fields are now missing: ```Were there any threats to independence noted*``` and ```Verification conclusion*``` **same thing when I get out of the form and go back to the url, the values I saved will be there but these fields are now missing**
<img width="969" alt="Image" src="https://github.com/user-attachments/assets/647d19a7-3b65-4639-b58c-b267ddcf47a3" />

**Comment by fviduya** [comment]
I have retested this in the DEV environment. [Here's](https://bcgov-my.sharepoint.com/:w:/g/personal/francesca_viduya_gov_bc_ca/EfG9lWUrSMtFkMiOPX3imQwBCWVD1cqr-3iBB5G7mNubwA?e=opPP8e) the test document for reference.

Just some small notes on this:
- Figma shows the field name should be ```Site visited```, but it currently shows ```Sites visited```
- Figma shows buttons should be Back and Save on the left side, and Continue on the right side, it currently shows Save & Continue on the right side - this functions properly though

In my previous test, the fields ```Were there any threats to independence noted*``` and ```Verification conclusion*``` gets lost when clicking Verification task from Attachments page. In my current test, that is not happening anymore. 

cc: @vesselak / @patriciarussellCAS
